### PR TITLE
executor: Change interface to datum in executor.

### PR DESCRIPTION
**Pull request by zxylvlp**

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: util/types/datum.go [hunk 1]
```diff
@@ -1447,3 +1448,39 @@ func EqualDatums(a []Datum, b []Datum) (bool, error) {
...
+// DatumEquals checker.
+type datumEqualsChecker struct {
+	*check.CheckerInfo
+}
+
+// DatumEquals checker verifies that the obtained value is equal to
+// the expected value.
+// For example:
+//     c.Assert(value, DatumEquals, NewDatum(42))
+var DatumEquals check.Checker = &datumEqualsChecker{
```
Comment (shenli) [inline util/types/datum.go]: This is used in test case only. Why move it here?

Comment (zxylvlp) [inline util/types/datum.go]: Because there are two package need to use it, but it can't expose in *_test.go .

File: executor/executor_simple.go [hunk 2]
```diff
@@ -188,7 +196,7 @@ func (e *SimpleExec) executeSetCharset(s *ast.SetCharsetStmt) error {
 
 func (e *SimpleExec) executeDo(s *ast.DoStmt) error {
 	for _, expr := range s.Exprs {
-		_, err := evaluator.Eval(e.ctx, expr)
```
Comment (shenli) [inline executor/executor_simple.go]: Can we drop original Eval and rename EvalDatum to Eval?

File: util/types/datum.go (hunk 1 above)
Comment (coocood) [inline util/types/datum.go]: We should avoid import test packages in non-test code. because it generates a lot of test flags in the executable file.

**Comment by zxylvlp** [comment]
@shenli  PTAL

**Comment by shenli** [comment]
LGTM
Well done!

**Pull request merged**